sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogRecordData.java

```diff
@@ -42,15 +43,23 @@ static SdkLogRecordData create(
         spanContext,
         severity,
         severityText,
-        attributes,
-        totalAttributeCount,
-        body);
+        0,
+        body,
+        attributes);
   }
 
   @Override
   @Nullable
   public abstract Value<?> getBodyValue();
 
+  @Override
+  public abstract ExtendedAttributes getExtendedAttributes();
+
+  @Override
+  public Attributes getAttributes() {
+    return getExtendedAttributes().asAttributes();
```

Comment:
the copy isn't ideal from perf perspective, though maybe it could be optimized by returning a filtered view into the underlying extended attributes struct somehow

it seems like a good compromise from backcompat perspective

Reply:
> though maybe it could be optimized by returning a filtered view into the underlying extended attributes struct somehow

Yeah I think there's some potential here. One challenge (as we saw with [FilteredAttributes](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/FilteredAttributes.java#L210-L265)) is having an equals implementation that works against the standard Attributes implementation.

Ideally, all callers migrate to calling `getExtendedAttributes`, which allows them to get around the performance hit. My intuition tells me there's not too many callers of this today.

Reply:
> Ideally, all callers migrate to calling `getExtendedAttributes`, which allows them to get around the performance hit. My intuition tells me there's not too many callers of this today.

yeah, this makes sense to me. maybe just adding a javadoc warning on getAttributes (or even eventually deprecating it?) to encourage users to move away from it